pkg/cmd/cli/describe/chaindescriber.go

```diff
@@ -0,0 +1,191 @@
+	}
+	loadingFuncs := []func() error{}
+	for _, loader := range loaders {
+		loadingFuncs = append(loadingFuncs, loader.Load)
+	}
+
+	if errs := parallel.Run(loadingFuncs...); len(errs) > 0 {
+		return g, utilerrors.NewAggregate(errs)
+	}
+
+	for _, loader := range loaders {
+		loader.AddToGraph(g)
+	}
+
+	buildedges.AddAllInputOutputEdges(g)
+
+	return g, nil
+}
+
+// Describe returns the output of the graph starting from the provided
+// image stream tag (name:tag)
+func (d *ChainDescriber) Describe(namespace, name, tag string) (string, error) {
+	g, err := d.MakeGraph(d.namespaces)
+	if err != nil {
+		return "", err
+	}
+
+	// Retrieve the imageStreamTag node of interest
+	istNode := g.Find(imagegraph.ImageStreamTagNodeName(imagegraph.MakeImageStreamTagObjectMeta(namespace, name, tag)))
+	if istNode == nil {
+		return "", NotFoundErr(fmt.Sprintf("%s:%s", name, tag))
+	}
+
+	// Partition down to the subgraph containing the ist of interest
+	partitioned := partition(g, istNode)
+
+	switch d.output {
+	case "dot":
+		data, err := dot.Marshal(partitioned, validDot(fmt.Sprintf("%s:%s", name, tag)), "", "  ", false)
+		if err != nil {
+			return "", err
+		}
+		return string(data), nil
+	case "":
+		return d.humanReadableOutput(partitioned, istNode), nil
+	}
+
+	return "", fmt.Errorf("unknown specified format %q", d.output)
+}
+
+// partition the graph down to a subgraph starting from the given root
+func partition(g osgraph.Graph, root graph.Node) osgraph.Graph {
+	// Filter out all but BuildConfig and ImageStreamTag nodes
+	nodeFn := osgraph.NodesOfKind(buildgraph.BuildConfigNodeKind, imagegraph.ImageStreamTagNodeKind)
+	// Filter out all but BuildInputImage and BuildOutput edges
+	edgeFn := osgraph.EdgesOfKind(buildedges.BuildInputImageEdgeKind, buildedges.BuildOutputEdgeKind)
+	sub := g.Subgraph(nodeFn, edgeFn)
+
+	// Filter out inbound edges to the ist of interest
+	edgeFn = osgraph.RemoveInboundEdges([]graph.Node{root})
+	sub = sub.Subgraph(nodeFn, edgeFn)
+
+	// Check all paths leading from the root node, collect any
+	// node found in them, and create the desired subgraph
+	desired := []graph.Node{root}
+	paths := path.DijkstraAllPaths(sub)
+	for _, node := range sub.Nodes() {
+		if node == root {
+			continue
+		}
+		path, _, _ := paths.Between(root, node)
+		if len(path) != 0 {
+			desired = append(desired, node)
+		}
+	}
+	return sub.SubgraphWithNodes(desired, osgraph.ExistingDirectEdge)
+}
+
+// humanReadableOutput traverses the provided graph using DFS and outputs it
+// in a human-readable format. It starts from the provided root, assuming it
+// is an imageStreamTag node and continues to the rest of the graph handling
+// only imageStreamTag and buildConfig nodes.
+func (d *ChainDescriber) humanReadableOutput(g osgraph.Graph, root graph.Node) string {
+	var singleNamespace bool
+	if len(d.namespaces) == 1 && d.namespaces[0] != kapi.NamespaceAll {
+		singleNamespace = true
+	}
+	depth := map[graph.Node]int{
```

Comment:
We don't have source nodes in the graph but yes, a tag for example can be in multiple depths. This is a cyclic dependency. I guess we should warn about cyclic dependencies in `build-chain` too. As for this, I am not sure how to handle it. Output the tag in both places? Need to think about it.